dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -117,6 +118,10 @@ private void parseRecursive(Reader source, Listener listener) throws IOException
                 break;
             }
             buffer.add((char) readCharacter);
+            if (getStringRepresentation(buffer).contains("{{")) {
```

Comment:
Did you meant startsWith ?

Reply:
In case there are some words before the {{.}} macro then startsWith() may limit it.